### DISCUSSION TITLE
I am new to your chain. What is a new item you want to see?

### DISCUSSION THREAD

**Discussion by CryptoHighway84** [OAK-Foundation/OAK-blockchain#316] [Help] [answered]
Good Day, 

I am interested in experimenting with this chain, and make a demo build, to see what I can come up with for the weekend. Are there any projects, needs, or developments, apps you would want to see in this community. As I learn more, I will get a feel for the system, I am excited to learn more! 

-Jade

**Comment by mattOAK** [comment, marked as answer]
Welcome! Here are some examples of use cases that we're excited to see built using OAK:
- Scheduled and recurring payments
- Dollar-cost averaging
- Wallet balance top-ups
More details on these examples are available on the OAK documentation page:
https://docs.oak.tech/docs/examples/

**Reply by CryptoHighway84** [reply]
//Here we have a smart contract that can help with waller top-ups
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.0;

contract WalletBalanceTopUp {
    address payable private walletAddress;  // the wallet address to top-up
    uint256 private topUpAmount;  // the amount to top-up
    uint256 private topUpInterval;  // the interval between top-ups
    uint256 private lastTopUpTime;  // the timestamp of the last top-up
    
    constructor(address payable _walletAddress, uint256 _topUpAmount, uint256 _topUpInterval) {
        walletAddress = _walletAddress;
        topUpAmount = _topUpAmount;
        topUpInterval = _topUpInterval;
        lastTopUpTime = block.timestamp;  // initialize the last top-up time to contract deployment time
    }
    
    function topUpWallet() public payable {
        require(msg.sender == address(this), "Only the contract can call this function");
        require(block.timestamp >= lastTopUpTime + topUpInterval, "Top-up interval not elapsed yet");
        require(address(this).balance >= topUpAmount, "Insufficient contract balance for top-up");
        
        // transfer top-up amount to wallet address
        (bool success,) = walletAddress.call{value: topUpAmount}("");
        require(success, "Failed to transfer top-up amount to wallet");
        
        // update last top-up time
        lastTopUpTime = block.timestamp;
    }
}

**Comment by yhl1998** [comment]
您的邮件已收到，我会尽快查收，谢谢。                                                      ——杨贺淋
